### PR TITLE
Webform submission encryption

### PR DESCRIPTION
**Pull request by cableman**
Added web form encrypt module and created helper module to toggle encryption.

### CODE REVIEW

File: modules/os2forms_encrypt/src/Form/SettingsForm.php [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    $config = $this->config('os2forms_encrypt.settings');
+
+    $link = Link::createFromRoute($this->t('administration'), 'entity.key.collection');
+    $form['notice'] = [
+      '#type' => 'inline_template',
+      '#template' => '<h3>{{ title }}</h3><p>{{ message|t }}</p><p>{{ adminMessage|t }}</p>',
+      '#context' => [
+        'title' => 'Please note',
+        'message' => 'The encryption key that comes with this module should <strong>not</strong> be used and should be changed before encrypting anything.',
+        'adminMessage' => 'You can modify the key (named "webform") in the keys ' . $link->toString() . ' panel. Additionally, the execution of this command can generate a new 256-bit key for you: <pre>dd if=/dev/urandom bs=32 count=1 | base64 -i -</pre>',
```
Comment (jekuaitk) [inline modules/os2forms_encrypt/src/Form/SettingsForm.php:48]: Should we allow translation of these?

File: modules/os2forms_encrypt/src/Element/WebformElementEncrypt.php [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\os2forms_encrypt\Form\SettingsForm;
+
+/**
+ * Class WebformElementEncrypt.
+ *
+ * This class contains a static method to process element attributes for
+ * webforms.
+ *
+ * @package Drupal\Core\Extension\Module\YourModule\WebformElementEncrypt
```
Comment (jekuaitk) [inline modules/os2forms_encrypt/src/Element/WebformElementEncrypt.php:14]: Should this be updated?

File: modules/os2forms_encrypt/src/Form/SettingsForm.php [hunk 3]
```diff
@@ -0,0 +1,73 @@
+<?php
+
+namespace Drupal\os2forms_encrypt\Form;
+
+use Drupal\Core\Form\ConfigFormBase;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Link;
+
+/**
+ * Class Os2FormsEncryptAdminForm.
```
Comment (jekuaitk) [inline modules/os2forms_encrypt/src/Form/SettingsForm.php:10]:
```suggestion
 * Class SettingsForm.
```

**Review by jekuaitk — CHANGES_REQUESTED** [review]

File: modules/os2forms_encrypt/src/Form/SettingsForm.php (hunk 1 above)
Comment (cableman) [reply modules/os2forms_encrypt/src/Form/SettingsForm.php:48]: They are translated in the inline template above where they er twig pipe into `t` :-)

File: modules/os2forms_encrypt/src/Form/SettingsForm.php (hunk 3 above)
Comment (cableman) [reply modules/os2forms_encrypt/src/Form/SettingsForm.php:10]: Fixed

File: modules/os2forms_encrypt/src/Element/WebformElementEncrypt.php (hunk 2 above)
Comment (cableman) [reply modules/os2forms_encrypt/src/Element/WebformElementEncrypt.php:14]: Removed

File: modules/os2forms_encrypt/src/Form/SettingsForm.php [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+  /**
+   * {@inheritdoc}
+   */
+  public function buildForm(array $form, FormStateInterface $form_state): array {
+    $config = $this->config('os2forms_encrypt.settings');
+
+    $link = Link::createFromRoute($this->t('administration'), 'entity.key.collection');
+    $form['notice'] = [
+      '#type' => 'inline_template',
+      '#template' => '<h3>{{ title }}</h3><p>{{ message|t }}</p><p>{{ adminMessage|t }}</p>',
```
Comment (jekuaitk) [inline modules/os2forms_encrypt/src/Form/SettingsForm.php:46]:
```suggestion
      '#template' => '<h3>{{ title|t }}</h3><p>{{ message|t }}</p><p>{{ adminMessage|t }}</p>',
```

File: modules/os2forms_encrypt/src/Form/SettingsForm.php (hunk 1 above)
Comment (jekuaitk) [reply modules/os2forms_encrypt/src/Form/SettingsForm.php:50]: Oh, i missed that

**Review by jekuaitk — APPROVED** [review]

**Review by thollebellcom — APPROVED** [review]

**Pull request merged**